region/maridia/inner-pink/Botwoon Energy Tank Room.json

```diff
@@ -220,13 +222,154 @@
       "name": "Base",
       "requires": []
     },
+    {
+      "link": [1, 2],
+      "name": "G-Mode",
+      "notable": false,
+      "entranceCondition": {
+        "comeInWithGMode": {
+          "mode": "any",
+          "morphed": false
+        }
+      },
+      "requires": [],
+      "exitCondition": {
+        "leaveWithGMode": {
+          "morphed": false
+        }
+      }
+    },
+    {
+      "link": [1, 2],
+      "name": "G-Mode Morph",
+      "notable": false,
+      "entranceCondition": {
+        "comeInWithGMode": {
+          "mode": "any",
+          "morphed": true
+        }
+      },
+      "requires": [
+        {"or": [
+          "Bombs",
+          {"and": [
+            {"ammo": {"type": "PowerBomb", "count": 2}},
+            {"or": [
+              "canTrickyJump",
+              "SpringBall",
+              "Gravity",
+              {"ammo": {"type": "PowerBomb", "count": 1}}
+            ]}
+          ]},
+          {"and": [
+            "Gravity",
+            {"enemyDamage": {
+              "enemy": "Puyo",
+              "type": "contact",
+              "hits": 1
+            }}
+          ]},
+          {"and": [
+            "SpringBall",
+            {"enemyDamage": {
+              "enemy": "Puyo",
+              "type": "contact",
+              "hits": 3
+            }}
+          ]},
+          {"enemyDamage": {
+            "enemy": "Puyo",
+            "type": "contact",
+            "hits": 7
+          }}
+        ]}
+      ],
+      "exitCondition": {
+        "leaveWithGMode": {
+          "morphed": true
+        }
+      }
+    },
     {
       "link": [1, 3],
       "name": "Base",
       "requires": [
         "h_canNavigateUnderwater"
       ]
     },
+    {
+      "link": [1, 3],
+      "name": "G-Mode",
+      "notable": false,
+      "entranceCondition": {
+        "comeInWithGMode": {
+          "mode": "any",
+          "morphed": false
+        }
+      },
+      "requires": [],
+      "exitCondition": {
+        "leaveWithGMode": {
+          "morphed": false
+        }
+      }
+    },
+    {
+      "link": [1, 3],
+      "name": "G-Mode Morph",
+      "notable": false,
+      "entranceCondition": {
+        "comeInWithGMode": {
+          "mode": "any",
+          "morphed": true
+        }
+      },
+      "requires": [
+        {"or": [
+          "Bombs",
```

Comment:
I think you need Grav or Spring to reach 3.  Maybe you could damage boost across with the Puyos.